### PR TITLE
ABLDuck - Documentation of parent class

### LINKED ISSUE #387: ABLDuck - Problem with hierarchie
### Subject of the issue
There is a problem with the hierarchie Tree

### Your environment
* Win10
* OpenEdge 11.7.4
* Ant 10
* PCT 213

### Actual behaviour
When my class is like that, I can't see parents methods in documentation :  
`CLASS Hello.Son INHERITS Father:`

But if my class is like that, it works fine : 
`CLASS Hello.Daughter INHERITS Hello.Father:`

But, fun fact, the `Hello.Father` can see his 2 daughter class

### Expected behaviour
The 2 way should work to declare a class.

### Steps to reproduce
See https://github.com/clement-brodu/OEdoc-sample

Thanks

### PR DESCRIPTION
**Pull request by clement-brodu**
# Description

I replaced short classes names by full classes names.

Fixes #387 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My branch is started from the latest commit in master
- [x] My commit history is clean
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/java/za/co/mip/ablduck/ABLDuck.java [hunk 1]
```diff
@@ -290,13 +295,28 @@ public void execute() {
 
             // Add implementers to the interface
             if (cu.implementations != null && !cu.implementations.isEmpty()) {
-                for (String i : cu.implementations) {
-                    String fullInterfacePath = determineFullyQualifiedClassName(cu.uses, i);
+                // TempList to avoid conflict
+                List<String> listNewImplement = new ArrayList<>();
+                Iterator<String> iter = cu.implementations.iterator();
+                
+                while(iter.hasNext()) {
```
Comment (gquerret) [inline src/java/za/co/mip/ablduck/ABLDuck.java]: Use foreach syntax pleae

File: src/java/za/co/mip/ablduck/ABLDuck.java [hunk 2]
```diff
@@ -290,13 +295,28 @@ public void execute() {
...
+                
+                while(iter.hasNext()) {
+                    String currentImplement = iter.next();
+                    String fullInterfacePath = determineFullyQualifiedClassName(cu.uses, currentImplement);
                     CompilationUnit iface = classes.get(fullInterfacePath);
                     if (iface != null) {
                         iface.implementers.add(cu.name);
+                        // Set FullName for implemented interfaces
+                        if (!currentImplement.equals(fullInterfacePath)){
+                            iter.remove();
```
Comment (gquerret) [inline src/java/za/co/mip/ablduck/ABLDuck.java:331]: Did you test that ? Removing elements in an iterator in the middle of a loop with next() is usually a bad idea.

Comment (clement-brodu) [reply src/java/za/co/mip/ablduck/ABLDuck.java:331]: Yes, it works fine.  
I tested it on my project and I had no problem.

It's why I use an iterator that allow me to remove items safely (unlike a simple foreach). 
But if you don't like it I can find another way (a foreach on a temp list should work as well)

Comment (clement-brodu) [reply src/java/za/co/mip/ablduck/ABLDuck.java:331]: I found the trick here : https://stackoverflow.com/questions/1196586/calling-remove-in-foreach-loop-in-java

Comment (gquerret) [reply src/java/za/co/mip/ablduck/ABLDuck.java:331]: I didn't write the ABLDuck code myself, but this block code is just supposed to replace the interface names with their fully-qualified name ? Could you just add a test case for that ? Code coverage says that this loop is not even tested.

Comment (clement-brodu) [reply src/java/za/co/mip/ablduck/ABLDuck.java:331]: I had the test case.

**Pull request merged**